lib/api/multiObjectDelete.js

```diff
@@ -128,48 +200,93 @@ export function getObjMetadataAndDelete(bucketName, quietSetting,
 
     // doing 5 requests at a time. note that the data wrapper
     // will do 5 parallel requests to data backend to delete parts
-    return async.forEachLimit(inPlay, 5, (key, moveOn) => {
-        metadata.getObjectMD(bucketName, key, {}, log, (err, objMD) => {
+    return async.forEachLimit(inPlay, 5, (entry, moveOn) => {
+        const opts = {
+            versionId: (entry.versionId === 'null' ? '' : entry.versionId),
```

Comment:
its because metadata does not know about the null version, so we have to get the master version to know where the null version is :)